File: dotnet/RelevantCodes.ExtentReports/RelevantCodes.ExtentReports/Model/Log.cs
using System;

namespace RelevantCodes.ExtentReports.Model
{
    public class Log
    {
        public DateTime Timestamp { get; set; }
        public LogStatus LogStatus { get; set; }
        public string StepName { get; set; }
        public string Details { get; set; }
    }
}

File: ruby/lib/extent/model/log.rb
require 'rubygems'
require 'liquid'

require_relative '../icon'

module RelevantCodes
	module Model
		class Log < Liquid::Drop
			attr_accessor :step_name, :details
			attr_writer :status
			attr_reader :timestamp
						
			def initialize
				@timestamp = Time.new.strftime(TIME_FORMAT)
			end
			
			def status
				return @status.to_s
			end
			
			def icon
				RelevantCodes::Icon.icon(status)
			end
		end
	end
end
